_includes/includes/kb_doc_template.php

```diff
@@ -158,27 +164,48 @@ function write_version_history($pagename){
       }
       $i++;
     }
-    $string.= '</ul><br>';
+    $string.= '</ul>';
+    return $string;
+  }
+
+  /* 
+    This functions works to show the keyboard download link
+    on the keyboard help documentation of help.keyman.com
+  */
+  function download_keyboard($pagename) {
+    global $keyboardname, $kbdid; // from head() function
+    
+    if(empty($kbdid)) return '';
+    
+    if(empty($pagename)) $pagename = '';
+    
+    $string = '<h2>Download this keyboard</h2><ul id="download-keyboard">';
+            
+    $string.= "<li><a href='{$KeymanHosts->keyman_com}/keyboards/$kbdid'>Download $keyboardname</a></li>";
+            
+    $string.= '</ul>';
+    
     return $string;
   }
 
   function begin_main($pagename){
-    global $version_history;
+    global $version_history, $download;
     write_breadcrumbs();
     $version_history = write_version_history($pagename);
+    $download = download_keyboard($pagename);
```

Comment:
The variable name `$download` is a little bit too generic (`$version_history` is also not great but that's a separate problem).

Can you call it `$download_link_html` -- that will be less likely to be a problem in the future.

```suggestion
    $download_link_html = build_download_keyboard_link_html();
```